src/components/ConfirmedRoute.tsx

```diff
@@ -98,7 +102,9 @@ function ConfirmedRoute({mapboxAccessToken, transaction, isSmallerIcon, shouldHa
         return MapboxToken.stop;
     }, []);
 
-    return !isOffline && Boolean(mapboxAccessToken?.token) ? (
+    const shouldDisplayMap = !shouldDisplayMapOnlyIfCoordinatesExist || (shouldDisplayMapOnlyIfCoordinatesExist && !!coordinates.length);
```

Comment:
The logic here is redundant/overly complicated. This is more simple and equivalent.

```
const shouldDisplayMap = !shouldDisplayMapOnlyIfCoordinatesExist || !!coordinates.length;
```